libs/java/auth_core/src/main/java/com/yahoo/athenz/auth/impl/CertificateAuthority.java

```diff
@@ -116,6 +120,11 @@ public Principal authenticate(X509Certificate[] certs, StringBuilder errMsg) {
 
         // create principal
         X509Certificate x509Cert = certId.getX509Certificate();
+
+        if (issuerDNs != null && !issuerDNs.isEmpty() && !issuerDNs.contains(Crypto.extractIssuerDn(x509Cert))) {
```

Comment:
Rather than having this functionality here, have you looked at the CertificateIdentityParser class. I expect the extra validation of issuer DN to happen within this class and not anywhere else.